test/EngineTest.cpp

```diff
@@ -69,7 +69,7 @@ void testOptionalJoin(const IdTable& inputA, const IdTable& inputB,
                    makeAllocator()};
     // Join a and b on the column pairs 1,2 and 2,1 (entries from columns 1 of
     // a have to equal those of column 2 of b and vice versa).
-    OptionalJoin::optionalJoin(inputA, inputB, jcls, &result);
+    OptionalJoin{}.optionalJoin(inputA, inputB, jcls, &result);
```

Comment:
I understand your intentions.
In the `Join` class I think we have tagged this constructor explicitly with an `OnlyForTesting` tag. Andd you should at least pass an `executionContext` which is easy via the `ad_utility::testing::getQec()` helper function.